src/qt/clientmodel.cpp

```diff
@@ -116,9 +116,22 @@ int ClientModel::getNumBlocks() const
 
 uint256 ClientModel::getBestBlockHash()
 {
+    // Lock order must be: first `cs_main`, then `m_cached_tip_mutex`.
+
+    uint256 tip{WITH_LOCK(m_cached_tip_mutex, return m_cached_tip_blocks)};
+
+    if (!tip.IsNull()) {
+        return tip;
+    }
+
+    tip = m_node.getBestBlockHash(); // Will lock `cs_main` (and release it).
```

Comment:
```suggestion
    tip = m_node.getBestBlockHash(); // Will lock `cs_main` (and release it). This is needed to be done early to not violate the implicit lock order cs_main -> m_cached_tip_mutex
```

I think the comment about the lock order can be combined with this comment?

Reply:
Combined the two comments and reworded them a little.